src/opm/output/eclipse/WriteInit.cpp

```diff
@@ -559,6 +558,63 @@ namespace {
 
         initFile.write("TRANNNC", singlePrecision(tran));
     }
+
+    // output aquifer cell and aquifer connection information for numerical aquifers
+    void writeNumericalAquifers(const Opm::NumericalAquifers& num_aquifers,
+                                const ::Opm::EclipseGrid&          grid,
+                                ::Opm::EclIO::OutputStream::Init&  initFile)
+    {
+        std::vector<int> aquifern(grid.getNumActive(), 0);
+        // aquifer cells
+        const auto& aquifer_cells = num_aquifers.allAquiferCells();
+        for ([[maybe_unused]] const auto& [cell_idx, cell] : aquifer_cells) {
+            const size_t active_index = grid.activeIndex(cell->global_index);
+            aquifern[active_index] -= std::exp2(cell->aquifer_id - 1);
+        }
+
+        // aquifer connections
+        for (const auto& [id, aqu] : num_aquifers.aquifers()) {
+            const auto& connections = aqu.connections();
+            const int exp2_id_1 = std::exp2(id - 1);
+            for (const auto& con : connections) {
+                const size_t active_index = grid.activeIndex(con.global_index);
+                aquifern[active_index] += exp2_id_1;
```

Comment:
Do we need to cater to the possibility that a single cell is connected to multiple aquifers?  If not, I'd prefer to just use regular assignment here.

Reply:
> Do we need to cater to the possibility that a single cell is connected to multiple aquifers? 

Yes, it looks like a single cell can be connected to multiple aquifers. If other implementation does not allow the aquifers to do so, we need to correct them. But here, let us keep the support.